src/include/txn_inline.h

```diff
@@ -570,10 +578,9 @@ __wt_txn_visible_all(WT_SESSION_IMPL *session, uint64_t id, wt_timestamp_t times
         return (true);
 
     /* If no oldest timestamp has been supplied, updates have to stay in cache. */
-    if (S2C(session)->txn_global.has_pinned_timestamp) {
-        __wt_txn_pinned_timestamp(session, &pinned_ts);
-        return (timestamp <= pinned_ts);
-    }
+    __wt_txn_pinned_timestamp(session, &pinned_ts);
+    if ((pinned_ts > 0U) && (timestamp <= pinned_ts))
```

Comment:
this could instead be:

```
if (pinned_ts != WT_TS_NONE && timestamp <= pinned_ts)
    return true;
```

Etienne, though you asked me on slack, now that I see the statement from before, I don't mind having:

`return (pinned_ts != WT_TS_NONE && timestamp <= pinned_ts)`